xbmc/URL.cpp

```diff
@@ -412,6 +412,11 @@ void CURL::SetProtocolOptions(const std::string& strOptions)
   }
 }
 
+void CURL::SetPayload(const std::string& strPayload)
+{
+  m_strPayload = strPayload;
+}
```

Comment:
Just thought over it . I'll place it into CURLOptions but:

- Will remove CURL::get wich forms the URL with separators
- Will only support substitution of URL path in the direction URL with separators -> URL + filled URL members,
(for backwards compatibility

Then postdata is completely hidden for you.

OK?